experimental/dispatch_profiler/launchers.py

```diff
@@ -0,0 +1,194 @@
+    self.iree_benchmark_module_path = os.path.join(args.build_dir, 'tools',
+                                                   'iree-benchmark-module')
+
+    # path to iree-run-module tool. (for verification)
+    self.iree_run_module_path = os.path.join(args.build_dir, 'tools',
+                                             'iree-run-module')
+
+    # output vmfb files for verification and profiling.
+    split_k_suffix = "_".join(["split_k_slice", str(operation.split_k_slices)])
+
+    vmfb_verify_list = [ self.operation.name(), split_k_suffix, "verify.vmfb"] \
+      if operation.operation_kind == OperationKind.SplitkMatmul \
+      else [self.operation.name(), "verify.vmfb"]
+    vmfb_profile_list = [ self.operation.name(), split_k_suffix, "profile.vmfb"] \
+      if operation.operation_kind == OperationKind.SplitkMatmul \
+      else [self.operation.name(), "profile.vmfb"]
+
+    self.vmfb_verify_filepath = os.path.join(self.operation_path,
+                                             "_".join(vmfb_verify_list))
+    self.vmfb_benchmark_filepath = os.path.join(self.operation_path,
+                                                "_".join(vmfb_profile_list))
+
+    # reference implementation for the operation_kind.
+    self.reference_impl_map = {
+        OperationKind.Matmul: ReferenceMatmulOp,
+        OperationKind.SplitkMatmul: ReferenceMatmulOp,
+        OperationKind.BatchMatmul: ReferenceBatchMatmulOp,
+    }
+
+  def iree_compile(self, compilation_mode):
+    """Compiles the input mlir file to vmfb file."""
+
+    benchmark_dispatch_repeat_count = self.benchmark_dispatch_repeat_count if compilation_mode == CompilationMode.Profile else 1
+    vmfb_file = self.vmfb_benchmark_filepath if compilation_mode == CompilationMode.Profile else self.vmfb_verify_filepath
+
+    # Base iree-compile commandline
+    cmd = [self.iree_compile_path, self.source_mlir_file, "-o", f"{vmfb_file}"]
+
+    # General compilation options
+    cmd += [f"--iree-hal-target-backends={self.args.device}"]
+
+    if self.args.device == "cuda":
+      cmd += [f"--iree-hal-cuda-llvm-target-arch={self.args.cuda_arch}"]
+    if self.operation.operation_kind == OperationKind.SplitkMatmul:
+      cmd += [
+          f"--iree-flow-split-matmul-reduction={self.operation.split_k_slices}"
+      ]
+    if self.args.use_mma_sync:
+      cmd += [f"--iree-codegen-llvmgpu-use-mma-sync"]
+    if self.args.use_wmma:
+      cmd += [f"--iree-codegen-llvmgpu-use-wmma"]
+
+    # Compilation options for profiling
+    cmd += [
+        f"--iree-hal-benchmark-dispatch-repeat-count={benchmark_dispatch_repeat_count}"
+    ]
+
+    # Appends print ir options at the end of the command line.
+    if self.args.mlir_print_ir_after_all:
+      cmd += [f"--mlir-print-ir-after-all"]
+
+    if not os.path.exists(vmfb_file) or self.args.force_compile:
+      print(
+          f"[Compiling to ({CompilationModeNames[compilation_mode]})] {' '.join(cmd)}"
```

Comment:
I think it will also be convenient if we dump the cmd into a flagfile under the same directory. It might help manually reproduce single benchmark with some ad-hoc flags.